internal/ccsmp/ccsmp_helper.h

```diff
@@ -19,16 +19,23 @@
 
 #include "solclient/solClient.h"
 
+// Reexport error info fields as they need to be copied.
+// This struct allows detailed error info to be copied only when it needs to be, saving time and memory.
+typedef struct solClient_errorInfo_detailed
+{ 
+    solClient_subCode_t SubCode;
+    solClient_session_responseCode_t ResponseCode;
+    char ErrorStr[SOLCLIENT_ERRORINFO_STR_SIZE];
```

Comment:
This is the exact same error struct definition of the solClient_errorInfo_t no?
We do not need duplicated error type.
There is already a use of:
```
solClientErrorInfoPt := C.solClient_getLastErrorInfo()
```
Which should be a well known type. From solClient.h headers.
There is no need for a solClient_errorInfo_detailed_t type.